### PR TITLE
[AIRFLOW-867] Enable and fix lots of untested unit tests

### PR DESCRIPTION
**Pull request by gsakkis**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- *https://issues.apache.org/jira/browse/AIRFLOW-867*

[Master branch](https://travis-ci.org/apache/incubator-airflow/jobs/201242728): Ran 256 tests in 819.435s
[This branch](https://travis-ci.org/apache/incubator-airflow/jobs/201335186): Ran 360 tests in 912.790s
 
It also fixes a few subtle bugs not captured by the already enabled tests.

### CODE REVIEW

**Comment by jlowin** [comment]
Given the scope of this and the fact that it will impact many in-flight PRs that may need to restructure their unit tests, I think it should get attention quickly from a few sets of eyes. At first glance it looks good and the green light from travis is obviously encouraging. @bolkedebruin @r39132 @criccomini please take a look

ps @gsakkis nice work uncovering these sleeper tests :)

**Comment by bolkedebruin** [comment]
@gsakkis great stuff. Can you pease rebase and verify landscape? Might ask you to split it up, as this is a massive and we will indeed catch other changes mid-flight. It also becomes easier to track down issues if the changes are smaller.

**Comment by gsakkis** [comment]
@bolkedebruin the branch has been rebased. Although the (squashed) diff is big, (a) it is mostly due to renames and (b) almost all changes are in tests so the risk of introducing issues in the working code is minimal. Splitting it into smaller chunks may in fact lead to bugs by omission, e.g. fail to bring back all tests, so I would suggest merging it asap.

**Comment by bolkedebruin** [comment]
@gsakkis please make sure you squash the commits and relate the commit message to the Jira issue as per guidelines. I'll merge when rebased/squash and commit message is proper.

**Comment by gsakkis** [comment]
@bolkedebruin all done.

**Comment by bolkedebruin** [comment]
@aoen can you please have a look? Some signatures of functions have been changed.

File: airflow/models.py [hunk 1]
```diff
@@ -1108,7 +1108,6 @@ def are_dependencies_met(
         :param verbose: whether or not to print details on failed dependencies
         :type verbose: boolean
         """
-        dep_context = dep_context or DepContext()
```
Comment (aoen) [inline airflow/models.py:1111]: Why remove this line?

File: airflow/models.py [hunk 2]
```diff
@@ -1131,7 +1130,6 @@ def get_failed_dep_statuses(
             self,
             dep_context=None,
             session=None):
-        dep_context = dep_context or DepContext()
```
Comment (aoen) [inline airflow/models.py:1134]: Same as above (why remove this line?)

File: airflow/ti_deps/deps/base_ti_dep.py [hunk 3]
```diff
@@ -81,6 +81,10 @@ def get_dep_statuses(self, ti, session, dep_context):
         :param dep_context: the context for which this dependency should be evaluated for
         :type dep_context: DepContext
         """
+        from airflow.ti_deps.dep_context import DepContext
```
Comment (aoen) [inline airflow/ti_deps/deps/base_ti_dep.py:84]: We should avoid imports at local scope. I think the previous implementation was OK but it seems you have some concerns?

Comment (gsakkis) [reply airflow/ti_deps/deps/base_ti_dep.py:84]: The motivation is probably clearer in the [original commit](https://github.com/Avocarrot/incubator-airflow/commit/4653fd8b92fb5507a547318117e5d188b1f7e072) before the squash. While it's possible to make `dep_context` optional without a local import, it requires further refactoring and this PR is already big enough so I opted for the less invasive option. But feel free to refactor it or revert the previous signature (and update the dozens previously broken tests that were passing `None`) if you feel strongly about it.

**Comment by jlowin** [comment]
@aoen @bolkedebruin @gsakkis just want to touch base here -- as feared, we already have merge conflicts with new tests. If we want to commit this we will need to move rapidly.

**Comment by aoen** [comment]
I still find the local import thing and the signature changes pretty hacky personally, but I won't stop others from +1ing.

If my questions re the changed signature are answered and a TODO is added to get rid of the local module imports then I guess that's good enough for now since missing test coverage is a pretty scary thing.

**Comment by bolkedebruin** [comment]
I am all for increased/improved testing, but I want to see this PR being broken up. It touches several areas outside testing and this is where I feel uncomfortable. Such a large PR I am just not able to review properly.

**Comment by r39132** [comment]
@gsakkis Firstly, thanks for taking this on. It's been a while and the project has improved a lot since this time. If you'd like to reopen this PR, we will take an earnest look at it with a plan to merge quickly. I will close this PR for now and the associated JIRA. Pls reopen if you would like to revisit this work.

**Pull request closed without merging**